app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,36 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
```

Comment:
`show, edit, update, destroy` are not all allowed. Don't forget that only 3 of the HTTP method are allowed for this controller
That means some of those methods are not welcomed there.